dtm0/ut/main.c

```diff
@@ -105,7 +112,41 @@ static void dtm0_ut_send_fops(struct m0_rpc_session *cl_rpc_session)
 	M0_ASSERT(m0_dtm0_ts__invariant(&reply_data.dti_ts));
 
 	M0_UT_ASSERT(m0_dtm0_tid_cmp(&dcs, &txr.dtd_id, &reply_data) == M0_DTS_EQ);
+	m0_fop_put_lock(fop);
+
+
+	/* Test PERSISTENT message */
+
+	rc = m0_dtm0_tx_desc_init(&txr, 1);
+	M0_UT_ASSERT(rc == 0);
+	txr.dtd_ps.dtp_pa[0].p_fid = srv_dtm0_fid;
+	txr.dtd_ps.dtp_pa[0].p_state = M0_DTPS_PERSISTENT;
 
+	txr.dtd_ps.dtp_pa[0].p_fid = srv_dtm0_fid;
+	txr.dtd_id = (struct m0_dtm0_tid) {
+		.dti_ts = now,
+		.dti_fid = cli_srv_fid
+	};
+	fop = m0_fop_alloc_at(cl_rpc_session,
+			      &dtm0_req_fop_fopt);
+	req = m0_fop_data(fop);
+	req->dtr_msg = DTM_PERSISTENT;
+	req->dtr_txr = txr;
+
+	m0_mutex_lock(&log->dl_lock);
+	rc = m0_be_dtm0_log_update(log, NULL, &txr, &(struct m0_buf){});
```

Comment:
Why do we need to update the log here? shouldn't it be able to work in the case where Pmsg was received before the corresponding request was executed?
Also, is it safe to use NULL instead of tx here? I guess BE will cause SIGSEGV or panic.